catalog/spatial/geocoding/spatial-geocoding-lucene/src/main/java/org/codice/ddf/spatial/geocoding/query/GazetteerQueryCatalog.java

```diff
@@ -0,0 +1,290 @@
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryable;
+import org.codice.ddf.spatial.geocoding.context.NearbyLocation;
+import org.codice.ddf.spatial.geocoding.context.impl.NearbyLocationImpl;
+import org.locationtech.spatial4j.context.SpatialContext;
+import org.locationtech.spatial4j.shape.impl.PointImpl;
+import org.opengis.filter.Filter;
+import org.opengis.filter.sort.SortBy;
+import org.opengis.filter.sort.SortOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GazetteerQueryCatalog implements GeoEntryQueryable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(GazetteerQueryCatalog.class);
+
+  private static final SpatialContext SPATIAL_CONTEXT = SpatialContext.GEO;
+
+  private static final String ERROR_MESSAGE = "Unable to execute query on the catalog.";
+
+  private static final String SORT_BY = "additional.sort.bys";
+
+  private static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory();
+
+  private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTWriter::new);
+
+  private static final ThreadLocal<WKTReader> WKT_READER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTReader::new);
+
+  private static final int KM_TO_M = 1000;
+
+  private CatalogFramework catalogFramework;
+
+  private FilterBuilder filterBuilder;
+
+  private Filter tagFilter;
+
+  private List<Filter> featureCodeFilters;
+
+  public GazetteerQueryCatalog(CatalogFramework catalogFramework, FilterBuilder filterBuilder) {
+    this.catalogFramework = catalogFramework;
+    this.filterBuilder = filterBuilder;
+    this.featureCodeFilters = new ArrayList<>();
+
+    for (String cityFeatureCode : GeoCodingConstants.CITY_FEATURE_CODES) {
+      Filter filter =
+          filterBuilder
+              .attribute(GeoEntryMetacardType.FEATURE_CODE_ATTRIBUTE_NAME)
+              .is()
+              .equalTo()
+              .text(cityFeatureCode);
+      featureCodeFilters.add(filter);
+    }
+
+    tagFilter =
+        filterBuilder
+            .attribute(Core.METACARD_TAGS)
+            .is()
+            .like()
+            .text(GeoCodingConstants.DEFAULT_TAG);
+  }
+
+  @Override
+  public List<GeoEntry> query(String queryString, int maxResults) throws GeoEntryQueryException {
+    Filter textFilter = filterBuilder.attribute(Core.TITLE).is().like().text(queryString);
+    Filter queryFilter = filterBuilder.allOf(tagFilter, textFilter);
+    Map<String, Serializable> properties = new HashMap<>();
+
+    SortBy featureCodeSortBy =
+        new SortByImpl(GeoEntryMetacardType.FEATURE_CODE_ATTRIBUTE_NAME, SortOrder.ASCENDING);
+    SortBy populationSortBy =
+        new SortByImpl(GeoEntryMetacardType.POPULATION_ATTRIBUTE_NAME, SortOrder.DESCENDING);
+    SortBy[] sortbys = {populationSortBy};
+    properties.put(SORT_BY, sortbys);
+
+    Query query = new QueryImpl(queryFilter, 1, maxResults, featureCodeSortBy, false, 10000L);
```

Comment:
Should `1000L` be a constant or configurable?

Reply:
made into a constant; in the future we can make it configurable if need be